src/lib/repositories/impl_oauth2/oauth2_repository_impl.py

```diff
@@ -0,0 +1,260 @@
+    SQL_QUERY_TO_GET_CLIENT_USER_BY_USER_NAME_AND_CLIENT_ID,
+)
+
+
+class Oauth2Repository:
+    @inject
+    def __init__(self, engine: Engine):
+        self.engine = engine
+        self.env_config = get_env_config_instance()
+
+    def login_client(self, client_id, client_secret):
+        return self._login(
+            client_id=client_id,
+            client_secret=client_secret,
+            grant_type=GranType.CLIENT_CREDENTIALS,
+        )
+
+    def login_user(self, client_id, client_secret, user_name, password):
+        return self._login(
+            user_name=user_name,
+            password=password,
+            client_id=client_id,
+            client_secret=client_secret,
+            grant_type=GranType.PASSWORD,
+        )
+
+    def _login(
+        self,
+        client_id=None,
+        client_secret=None,
+        user_name=None,
+        password=None,
+        grant_type=None,
+    ):
+        client_validation = self._validate_client_credentials(client_id, client_secret)
+
+        if client_validation:
+
+            client = self._get_client_by_client_id(client_id)
+
+            if grant_type == GranType.CLIENT_CREDENTIALS:
+
+                client_scope = self._get_scope_by_app_client_id(client.id)
+
+                access_token = build_client_credentials_access_token(
+                    client, client_scope.scope, self.env_config.secret_key
+                )
+                refresh_token = build_client_credentials_refresh_token(
+                    client, client_scope.scope, self.env_config.secret_key
+                )
+
+            elif grant_type == GranType.PASSWORD:
+                user_validation = self._validate_user_credentials(
+                    client, user_name, password
+                )
+                if user_validation:
+
+                    user = self._get_user_by_user_name(user_name)
+
+                    access_token = build_user_credential_access_token(
+                        user, client, self.env_config.secret_key
+                    )
+                    refresh_token = build_user_credential_refresh_token(
+                        user, client, self.env_config.secret_key
+                    )
+
+            persisted_refresh_token = self._add_refresh_token(
+                refresh_token, client.id, grant_type.value
+            )
+            self._add_access_token(persisted_refresh_token.id, access_token)
+
+            return {"refresh_token": refresh_token, "access_token": access_token}
+
+    def validate_token(self, token):
+        try:
+            jwt.decode(token, self.env_config.secret_key, algorithms="HS256")
+            return True
+
+        except jwt.ExpiredSignatureError:
+            return False
+        except jwt.InvalidSignatureError:
+            return False
+        except jwt.DecodeError:
+            return False
+
+    def refresh_token(self, refresh_token):
+        refresh_token_validation = self.validate_token(refresh_token)
```

Comment:
No need to return anything from here, since an error should be thrown

Reply:
done.